File: backend/models/Order.js
const mongoose = require('mongoose');

const orderItemSchema = new mongoose.Schema({
  product: { type: mongoose.Schema.Types.ObjectId, ref: 'Product', required: true },
  vendor: { type: mongoose.Schema.Types.ObjectId, ref: 'Vendor', required: true },
  name: { type: String, required: true },
  image: { type: String },
  price: { type: Number, required: true },
  quantity: { type: Number, required: true, min: 1 },
});

const orderSchema = new mongoose.Schema({
  user: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true },
  orderNumber: { type: String, unique: true },
  items: [orderItemSchema],
  shippingAddress: {
    name: { type: String, required: true },
    street: { type: String, required: true },
    city: { type: String, required: true },
    state: { type: String, required: true },
    zip: { type: String, required: true },
    country: { type: String, default: 'US' },
  },
  paymentMethod: { type: String, enum: ['stripe', 'paypal'], required: true },
  paymentResult: {
    id: String,
    status: String,
    updateTime: String,
    email: String,
  },
  subtotal: { type: Number, required: true },
  tax: { type: Number, required: true, default: 0 },
  shippingCost: { type: Number, required: true, default: 0 },
  total: { type: Number, required: true },
  status: {
    type: String,
    enum: ['pending', 'confirmed', 'processing', 'shipped', 'delivered', 'cancelled', 'refunded'],
    default: 'pending',
  },
  isPaid: { type: Boolean, default: false },
  paidAt: { type: Date },
  isDelivered: { type: Boolean, default: false },
  deliveredAt: { type: Date },
  trackingNumber: { type: String },
  notes: { type: String },
}, { timestamps: true });

orderSchema.pre('save', function (next) {
  if (!this.orderNumber) {
    this.orderNumber = 'GOB-' + Date.now().toString(36).toUpperCase() + '-' + Math.random().toString(36).substring(2, 6).toUpperCase();
  }
  next();
});

module.exports = mongoose.model('Order', orderSchema);

File: ios/Packages/GOBShared/Sources/GOBShared/Models/Order.swift
import Foundation

public struct Order: Codable, Identifiable, Sendable {
    public let id: String
    public let userId: IntOrString?
    public let items: [OrderItem]
    public let shippingAddress: ShippingAddress?
    public let paymentMethod: String?
    public let subtotal: Double
    public let tax: Double
    public let shippingCost: Double
    public let total: Double
    public let status: String
    public let isPaid: Bool
    public let isDelivered: Bool
    public let paidAt: String?
    public let stripePaymentIntentId: String?
    public let refundStatus: String?
    public let refundedAmount: Double?
    public let createdAt: String?

    public var statusDisplay: String {
        switch status {
        case "pending": return "Pending"
        case "pending_payment": return "Awaiting Payment"
        case "confirmed": return "Confirmed"
        case "shipped": return "Shipped"
        case "delivered": return "Delivered"
        case "cancelled": return "Cancelled"
        default: return status.capitalized
        }
    }
}

public struct OrderItem: Codable, Sendable, Identifiable {
    public let productId: String
    public let name: String
    public let price: Double
    public let quantity: Int
    public let vendorId: String?

    public var id: String { productId }
}

public struct ShippingAddress: Codable, Sendable {
    public let street: String?
    public let city: String?
    public let state: String?
    public let zip: String?
    public let name: String?
    public let phone: String?

    public init(street: String, city: String, state: String, zip: String, name: String? = nil, phone: String? = nil) {
        self.street = street
        self.city = city
        self.state = state
        self.zip = zip
        self.name = name
        self.phone = phone
    }

    public var formatted: String {
        [street, city, state, zip].compactMap { $0 }.joined(separator: ", ")
    }
}
